tests/elster_xml/grundsteuer/test_elster_data_representation.py

```diff
@@ -7,12 +7,256 @@
+        flaechen = [59]
+
+        result = EWohnUnter60(flaechen)
+
+        assert result.E7403131 == 1
+        assert result.E7403132 == 59
+
+    def test_flaeche_60_should_retain_zeroes(self):
+        flaechen = [60]
+
+        result = EWohnUnter60(flaechen)
+
+        assert result.E7403131 == 0
+        assert result.E7403132 == 0
+
+    def test_flaechen_under_60_should_set_fields(self):
+        flaechen = [59, 1]
+
+        result = EWohnUnter60(flaechen)
+
+        assert result.E7403131 == 2
+        assert result.E7403132 == 60
+
+    def test_flaechen_one_under_60_should_set_fields(self):
+        flaechen = [59, 60]
+
+        result = EWohnUnter60(flaechen)
+
+        assert result.E7403131 == 1
+        assert result.E7403132 == 59
+
+
+class TestEWohn60bis100:
+    def test_flaeche_60_should_set_fields(self):
+        flaechen = [60]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 1
+        assert result.E7403142 == 60
+
+    def test_flaeche_99_should_set_fields(self):
+        flaechen = [99]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 1
+        assert result.E7403142 == 99
+
+    def test_flaeche_100_should_retain_zeroes(self):
+        flaechen = [100]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 0
+        assert result.E7403142 == 0
+
+    def test_flaechen_under_100_should_set_fields(self):
+        flaechen = [99, 60]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 2
+        assert result.E7403142 == 159
+
+    def test_flaechen_one_under_100_should_set_fields(self):
+        flaechen = [99, 100]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 1
+        assert result.E7403142 == 99
+
+
+class TestEWohnAb100:
+    def test_flaeche_100_should_set_fields(self):
+        flaechen = [100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 1
+        assert result.E7403152 == 100
+
+    def test_flaeche_99_should_retain_zeroes(self):
+        flaechen = [99]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 0
+        assert result.E7403152 == 0
+
+    def test_flaechen_from_100_should_set_fields(self):
+        flaechen = [100, 100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 2
+        assert result.E7403152 == 200
+
+    def test_flaechen_one_under_100_should_set_fields(self):
+        flaechen = [99, 100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 1
+        assert result.E7403152 == 100
+
+
+class TestEWeitereWohn:
+    def test_should_construct_from_dict_correctly(self):
+        weitere_wohnraeume = WeitereWohnraeumeDetails.parse_obj({"anzahl": 2, "flaeche": 42})
+
+        result = EWeitereWohn(weitere_wohnraeume)
+
+        assert result.E7403121 == 2
+        assert result.E7403122 == 42
```

Comment:
You dodn't want to include the len(vars()) test?

Reply:
No, for the same reasons we discussed privately. To summarize: any additional attributes set in the constructor will not be returned by `fields()` (which is used under the hood by `asdict()`) and hence will not end up in the XML. That was the only reason why I considered counting the variables useful before.

Reply:
Mh...But I could have by accident added an additional field to the class' definition and this test wouldn't catch that..

Reply:
Done.